test/Microsoft.ML.TestFramework/BaseTestBaseline.cs

```diff
@@ -594,28 +592,45 @@ protected bool CheckEqualityFromPathsCore(string relPath, string basePath, strin
             }
         }
 
-        private static void GetNumbersFromFile(ref string firstString, ref string secondString, decimal precision)
+        private static void GetNumbersFromFile(ref string firstString, ref string secondString, int digitsOfPrecision)
         {
             Regex _matchNumer = new Regex(@"\b[0-9]+\.?[0-9]*\b", RegexOptions.IgnoreCase | RegexOptions.Compiled);
             MatchCollection firstCollection = _matchNumer.Matches(firstString);
             MatchCollection secondCollection = _matchNumer.Matches(secondString);
 
-            MatchNumberWithTolerance(firstCollection, secondCollection, precision);
+            MatchNumberWithTolerance(firstCollection, secondCollection, digitsOfPrecision);
             firstString = _matchNumer.Replace(firstString, "%Number%");
             secondString = _matchNumer.Replace(secondString, "%Number%");
         }
 
-        private static void MatchNumberWithTolerance(MatchCollection firstCollection, MatchCollection secondCollection, decimal precision)
+        private static void MatchNumberWithTolerance(MatchCollection firstCollection, MatchCollection secondCollection, int digitsOfPrecision)
         {
             for (int i = 0; i < firstCollection.Count; i++)
             {
-                decimal f1 = decimal.Parse(firstCollection[i].ToString());
-                decimal f2 = decimal.Parse(secondCollection[i].ToString());
+                double f1 = double.Parse(firstCollection[i].ToString());
+                double f2 = double.Parse(secondCollection[i].ToString());
+
+                double allowedVariance = Math.Pow(10, -digitsOfPrecision);
+                double delta = Round(f1, digitsOfPrecision) - Round(f2, digitsOfPrecision);
 
-                Assert.InRange(f1, f2 - (f2 / precision), f2 + (f2 / precision));
+                Assert.InRange(delta, -allowedVariance, allowedVariance);
             }
         }
 
+        private static double Round(double value, int digitsOfPrecision)
+        {
+            if ((value == 0) || double.IsInfinity(value) || double.IsNaN(value))
```

Comment:
>(value == 0) [](start = 16, length = 12)

This is just me learning - I assume `value == 0` works for both zero and negative zero, right?

Reply:
Yes.